src_c/event.c

```diff
@@ -1365,6 +1365,54 @@ static PyMemberDef pg_event_members[] = {
     {NULL} /* Sentinel */
 };
 
+/* Deprecated joy attribute of event */
+static PyObject *
+pg_event_joy_get(pgEventObject *self, void *closure)
+{
+    PyObject *joy = PyDict_GetItemString(self->dict, "joy");
+    if (!joy) {
+        if (PyErr_Occurred()) {
+            return NULL;
+        }
+        return RAISE(PyExc_AttributeError,
+                     "'Event' object has no attribute 'joy'");
+    }
+    /* Don't warn if this is a custom event that happens to have an attribute
+     * named joy */
+    if (self->type < PGE_USEREVENT) {
+        if (PyErr_WarnEx(PyExc_DeprecationWarning,
+                         "the joy attribute is deprecated in favour of "
+                         "instance_ids, and "
+                         "will be removed in the future",
+                         1) != 0) {
+            return NULL;
+        }
+    }
+    Py_INCREF(joy);
```

Comment:
I'm not sure this is necessary. PyDict_GetItemString borrows the reference, and then if it gets put into a variable on the python side that would do the incref?

Reply:
This is necessary. On the python side, it is expected that a strong reference is always returned (so the incref does not happen automatically, the C extension programmer must take care of it)

Removing this will definitely cause problems and lead to segfaults in random places